ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Args.hs

```diff
@@ -50,12 +50,14 @@ data ChainDbArgs m blk = forall h1 h2 h3. ChainDbArgs {
       -- Decoders
       cdbDecodeHash       :: forall s. Decoder s (HeaderHash blk)
     , cdbDecodeBlock      :: forall s. Decoder s (Lazy.ByteString -> blk)
+    , cdbDecodeHeader     :: forall s. Decoder s (Lazy.ByteString -> Header blk)
     , cdbDecodeLedger     :: forall s. Decoder s (LedgerState blk)
     , cdbDecodeChainState :: forall s. Decoder s (ChainState (BlockProtocol blk))
 
       -- Encoders
-    , cdbEncodeBlock      :: blk -> BinaryInfo Encoding
     , cdbEncodeHash       :: HeaderHash blk -> Encoding
+    , cdbEncodeBlock      :: blk -> BinaryInfo Encoding
+    , cdbEncodeHeader     :: Header blk -> Encoding
```

Comment:
We should document that the encoder and decoder for headers include the envelope already. The chain DB will _only_ add the envelope if we are extracting the header from a block and it never _removes_ the envelope. This whole situation is a bit confusing, will be good to be quite explicit about it.